[docs] Add initial steps of creating a custom component (#2534)
mui/mui-toolpad: docs/data/toolpad/concepts/custom-components.md

<p class="description">You can bring your own custom components inside the Toolpad editor.</p>

The steps below explain how to create and use a custom component in Toolpad.

## Create component

> 1. Click **Create** in the custom component section of the Component library.

Reviewer: ```suggestion
1. Open the Component LIbrary by hovering over it with your mouse.
1. Click **Create** in the custom component section.
```

> 1. Start by clicking on Create in the custom component section of the Component library.

Reviewer 2: ```suggestion
1. Start by clicking on **Create** in the custom component section of the Component library.
```

> 2. Name your component.

Reviewer: ```suggestion
1. Supply a name for your component in the dialog that opens. Click **Create** to initialize a new component.
```

> 3. Open the snackbar prompt. It will open your code editor. In case it doesn't open, check [troubleshoot missing editor.](https://mui.com/toolpad/how-to-guides/editor-path/)

Reviewer: Let's add the note of the failing editor separately? Also punctuation should go outside of the link and we should let the markdown parser auto-number list items (all of them should start with `1.`)

```suggestion
1. In the snackbar that appears, click the **Open** button. Your code editor will open with the newly created component. 

::info
In case it doesn't open, check [troubleshoot missing editor].(https://mui.com/toolpad/how-to-guides/editor-path/)
::
```

> 3. Open the snackbar prompt, it will open up the code editor. In case it doesn't open, check [troubleshoot missing editor.](https://mui.com/toolpad/how-to-guides/editor-path/)

Reviewer 2: ```suggestion
3. Open the snackbar prompt. It will open your code editor. In case it doesn't open, check [troubleshoot missing editor.](https://mui.com/toolpad/how-to-guides/editor-path/)
```

Reviewer 2: Also we definitely should implement a more visible way for the user to start working on their new code component right away.

<video controls width="auto" height="100%" style="contain" alt="custom-component-creation">
  <source src="/static/toolpad/docs/concepts/custom-components/custom-component-creation.mp4" type="video/mp4">
  Your browser does not support the video tag.
</video>

## Add component code

> Toolpad exposes a `createComponent` API to bring in custom components to the Toolpad component library. For example, in `toolpad/components/HelloWorld.tsx`:

Reviewer: we should link `createComponent` to the relevant API doc

```suggestion
Toolpad exposes a [`createComponent`](<link goes here>) function. This is used to signal to Toolpad which are components that can be imported and how it should interpret the properties.
```

## `createComponent`

> Creating a custom component in Toolpad is a quick process:

Reviewer: Let's avoid language that describes steps as quick/easy/simple? (https://www.notion.so/Writing-style-guide-2a957a4168a54d47b14bae026d06a24b?d=bf8ae75e0eb94c96b6328a61367eb191&pvs=4#ad0d7e5fd0e6494a97f6febff143af96)

## Component Library

Custom components become available in the component library, alongside an option to create a new one.

> To use a custom component just drag it to the canvas.

Reviewer: Avoid language like "just", "simply", "easily"...
```suggestion
To use a custom component, drag it from the Component Library into the canvas. The component will be available under the **Custom Components** section.
```
